src/trace/context.ts

```diff
@@ -183,6 +213,22 @@ export function readTraceFromEvent(event: any): TraceContext | undefined {
   };
 }
 
+export function readTraceFromEvent(event: any): TraceContext | undefined {
+  if (typeof event !== "object") {
+    return;
+  }
+
+  if (event.headers && typeof event.headers === "object") {
```

Comment:
nit: I think you can just write `typeof event.headers === "object"`, since if event.headers is undefined, typeof event.headers === "undefined"